packages/canon/src/components/Menu/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Menu as MenuPrimitive } from '@base-ui-components/react/menu';
+
+/** @public */
+export type MenuComponent = {
```

Comment:
mkay loving the options here, will have to try it out :grin: 

We'll need a new sidebar component for the new frontend system and I'm thinking that might be the best place to start adopting Canon in core